src/encoding.md

```diff
@@ -688,6 +689,68 @@ fn run() -> Result<()> {
 # quick_main!(run);
 ```
 
+
+[ex-mime-from-filename]: #ex-mime-from-filename
+<a name="ex-mime-from-filename"></a>
+## Get MIME type from filename
+
+[![mime-badge]][mime] [![cat-encoding-badge]][cat-encoding]
+
+The following example shows how to return the correct MIME type from a given
+filename using the [mime] crate.
+
+```rust
+# #[macro_use]
+# extern crate error_chain;
+#[macro_use]
+
+use std::io;
+#
+# error_chain! {
+#    foreign_links {
+#        IOError(std::io::Error);
+#    }
+# }
+
+extern crate mime;
+use mime::Mime;
+
+/// Return the MIME type according to file extension. Default is text/plain
```

Comment:
This is a doc comment, it is useful for documenting modules, etc.  Regular comments have 2 slashes.  Rust Cookbook prefers descriptions over comments.  Please at minimum replace the doc comment with a comment.